### PR TITLE
Fix Welsh links on legacy pages

### PR DESCRIPTION
**Pull request by mattandrews**
Due to quirks of Sitecore, proxying and cookies, links on legacy pages between English/Welsh are sporadic in appearance. This code tweaks our proxying technique to append these links if found.

Some pages already have Welsh/English links and don't need them to be appended, eg:

- **Press release**: [English](https://www.biglotteryfund.org.uk/global-content/press-releases/wales/060318_wal_a4aw_release_march18) vs [Welsh](https://www.biglotteryfund.org.uk/welsh/global-content/press-releases/wales/060318_wal_a4aw_release_march18)
(eg. this content when proxied already has the links)

... but some content is available in Welsh but has no discernible link to switch language:

- **Research: Open Data**: [English](https://www.biglotteryfund.org.uk/research/open-data) vs [Welsh](https://www.biglotteryfund.org.uk/welsh/research/open-data)
(eg. this content when proxied does not have the links)

This PR checks to see whether the page is missing its opposite language link, and if so, adds it.

### CODE REVIEW

**Comment by mattandrews** [comment]
Hmm, confusingly, when I load the press release page above on the live site, I don't see the Cymraeg link, but I do see it on `TEST` and `DEV`. Possibly another cookie quirk / proxying server location issue, but I can confirm that locally, this code doesn't add a second language link when there's an existing one.

File: modules/legacy.js [hunk 1]
```diff
@@ -39,6 +40,41 @@ const proxyLegacyPage = ({ req, res, domModifications, followRedirect = true })
             dom.window.document.title = '[PROXIED] ' + titleText;
         }
 
+        const pageIsWelsh = isWelsh(req.path);
+        const hasWelshLink = dom.window.document.querySelectorAll('[hreflang="cy"]').length > 0;
```
Comment (davidrapson) [inline modules/legacy.js]: Worth scoping these selectors to `regionNav` too?

File: modules/legacy.js [hunk 2]
```diff
@@ -39,6 +40,41 @@ const proxyLegacyPage = ({ req, res, domModifications, followRedirect = true })
...
+        const hasEnglishLink = dom.window.document.querySelectorAll('[hreflang="en"]').length > 0;
+
+        const appendLanguageLink = localeToAppend => {
+            let linkPath, linkText;
```
Comment (davidrapson) [inline modules/legacy.js]: Maybe ternary statements for conciseness?

``` js
const linkPath = localeToAppend === 'cy' ? makeWelsh(req.path) : 'English';
const linkText = localeToAppend === 'cy' ? 'Cymraeg' : 'English';
```

**Comment by davidrapson** [comment]
👍

**Pull request merged**